src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/ProcessContext.cs

```diff
@@ -0,0 +1,217 @@
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+
+namespace LodeRunner.API.Test.IntegrationTests.ExecutingTestRun
+{
+    /// <summary>
+    /// Provides and Manages a Process Context.
+    /// </summary>
+    /// <seealso cref="System.IDisposable" />
+    internal class ProcessContext : IDisposable
+    {
+        private readonly ProcessContextParams processContextParams;
+        private readonly ITestOutputHelper output;
+
+        private bool disposedValue = false;
+        private Process currentProcess;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProcessContext"/> class.
+        /// </summary>
+        /// <param name="processContextParams">The Process Context Params.</param>
+        /// <param name="output">The output.</param>
+        public ProcessContext(ProcessContextParams processContextParams, ITestOutputHelper output)
+        {
+            this.processContextParams = processContextParams;
+            this.output = output;
+        }
+
+        /// <summary>
+        /// Gets the errors.
+        /// </summary>
+        /// <value>
+        /// The errors.
+        /// </value>
+        public List<string> Errors { get; private set; } = new List<string>();
+
+        /// <summary>
+        /// Gets the output.
+        /// </summary>
+        /// <value>
+        /// The output.
+        /// </value>
+        public List<string> Output { get; private set; } = new List<string>();
+
+        /// <summary>
+        /// Gets a value indicating whether this instance started.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if this instance started; otherwise, <c>false</c>.
+        /// </value>
+        public bool Started { get; private set; } = false;
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+            this.Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Starts the specified Process.
+        /// </summary>
+        /// <param name="delayReturn">delay Return (ms).</param>
+        /// <returns>whether or not the process started.</returns>
+        public bool Start(int delayReturn = 0)
+        {
+            if (!this.Started)
+            {
+                this.currentProcess = new Process
+                {
+                    StartInfo = new ProcessStartInfo()
+                    {
+                        CreateNoWindow = true,
+                        UseShellExecute = false,
+
+                        FileName = this.processContextParams.CommandLine,
+                        Arguments = this.BuildAndGetArguments(),
```

Comment:
It feels strange to have a BuildAndGetArguments to just concat a string for us. Will this function be reused?

Reply:
BuildAndGetArguments, also call other method, for readability purposes is a better approach to keep the method as it is instead, having a long concat string operation